hack/tools/Makefile

```diff
@@ -57,8 +57,8 @@ $(GO_APIDIFF): $(BIN_DIR) go.mod go.sum ## Build a local copy of go-apidiff.
 	go build -tags=tools -o $@ github.com/joelanford/go-apidiff
 
 GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
-$(GOLANGCI_LINT): $(BIN_DIR) go.mod go.sum ## Build a local copy of golangci-lint.
-	go build -tags=tools -o $@ github.com/golangci/golangci-lint/cmd/golangci-lint
+$(GOLANGCI_LINT): ## Download golangci-lint using hack script in the tools folder.
+	../ensure-golangci-lint.sh -b $(BIN_DIR)
```

Comment:
none of the other targets uses the scripts, any specific reason for using the script for this tool?

Reply:
The update of cluster-api module is `hack/tools/go.mod` was causing the downgrade of `golangci-lint `version. Since there is a hack script present, made this change for downloading the binary via the GitHub releases.

Reply:
why so? usually this should just stick to the version we specify in the go.mod file, lets debug this further on this.